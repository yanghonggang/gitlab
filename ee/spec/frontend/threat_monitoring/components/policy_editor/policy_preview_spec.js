import { GlTabs } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import PolicyPreview from 'ee/threat_monitoring/components/policy_editor/policy_preview.vue';

describe('PolicyPreview component', () => {
  let wrapper;

  const factory = ({ propsData } = {}) => {
    wrapper = shallowMount(PolicyPreview, {
      propsData: {
        ...propsData,
      },
    });
  };

  beforeEach(() => {
    factory({
      propsData: {
        policyYaml: 'foo',
        policyDescription: '<strong>bar</strong><br><div>test</div><script></script>',
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders policy preview tabs', () => {
    expect(wrapper.find(GlTabs).element).toMatchSnapshot();
  });

  describe('with initialTab', () => {
    beforeEach(() => {
      factory({
        propsData: {
          policyYaml: 'foo',
          policyDescription: 'bar',
          initialTab: 1,
        },
      });
    });

    it('selects initial tab', () => {
      expect(wrapper.find(GlTabs).attributes().value).toEqual('1');
    });
  });
});
