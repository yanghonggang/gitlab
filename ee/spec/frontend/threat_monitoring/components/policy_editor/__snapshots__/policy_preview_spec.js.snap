// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyPreview component renders policy preview tabs 1`] = `
<gl-tabs-stub
  contentclass="gl-pt-0"
  theme="indigo"
  value="0"
>
  <gl-tab-stub
    title=".yaml"
    titlelinkclass=""
  >
    <pre
      class="gl-bg-white gl-rounded-top-left-none gl-rounded-top-right-none"
    >
      foo
    </pre>
  </gl-tab-stub>
   
  <gl-tab-stub
    title="Rule"
    titlelinkclass=""
  >
    <div
      class="gl-bg-white gl-rounded-top-left-none gl-rounded-top-right-none gl-rounded-bottom-left-base gl-rounded-bottom-right-base gl-py-3 gl-px-4 gl-border-1 gl-border-solid gl-border-gray-100"
    >
      <strong>
        bar
      </strong>
      <br />
      test
    </div>
  </gl-tab-stub>
</gl-tabs-stub>
`;
